In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [2]:
df=pd.read_csv('train.csv')

In [3]:
df.head()

feature_1  feature_2  feature_3  feature_4  feature_5  feature_6  \
0   0.031138   0.079892   0.157382  -0.014636  -0.051778  -0.021332   
1   0.113040   0.175731   0.217741  -0.196254  -0.010129  -0.030586   
2   0.048570   0.091281   0.160776  -0.150937   0.020115   0.044117   
3   0.039212   0.118388   0.173831  -0.096659  -0.008702   0.061298   
4   0.056019   0.170639   0.157917  -0.228605  -0.065965  -0.088732   

   feature_7  feature_8  feature_9  feature_10  ...  feature_763  feature_764  \
0  -0.073593  -0.005386  -0.212557    0.099683  ...    -0.085248    -0.096007   
1   0.067114  -0.072412  -0.239192    0.104741  ...    -0.090283    -0.053885   
2  -0.050092  -0.045661  -0.155332    0.117206  ...    -0.021524    -0.008411   
3   0.008974  -0.003277  -0.065046    0.095480  ...    -0.071936    -0.023120   
4  -0.082243  -0.080568  -0.341500    0.142430  ...    -0.155621    -0.079447   

   feature_765  feature_766  feature_767  feature_768  label_1  label_2  \
0    -0.000766     0.021399    -0.041432     0.094806       45      NaN   
1    -0.010967     0.062209    -0.122958     0.192949       45      NaN   
2    -0.006248     0.031468    -0.056915     0.154731       45      NaN   
3    -0.007812     0.057600    -0.121892     0.072796       45      NaN   
4     0.015316     0.127726    -0.151966     0.169634       45      NaN   

   label_3  label_4  
0        1        6  
1        1        6  
2        1        6  
3        1        6  
4        1        6  

[5 rows x 772 columns]

In [4]:
##check null values
df.isnull().sum()

feature_1        0
feature_2        0
feature_3        0
feature_4        0
feature_5        0
              ... 
feature_768      0
label_1          0
label_2        480
label_3          0
label_4          0
Length: 772, dtype: int64

In [5]:
y_train = df[["label_1","label_2","label_3","label_4"]]
X_train = df.drop(y_train, axis=1)


In [6]:
X_train

feature_1  feature_2  feature_3  feature_4  feature_5  feature_6  \
0       0.031138   0.079892   0.157382  -0.014636  -0.051778  -0.021332   
1       0.113040   0.175731   0.217741  -0.196254  -0.010129  -0.030586   
2       0.048570   0.091281   0.160776  -0.150937   0.020115   0.044117   
3       0.039212   0.118388   0.173831  -0.096659  -0.008702   0.061298   
4       0.056019   0.170639   0.157917  -0.228605  -0.065965  -0.088732   
...          ...        ...        ...        ...        ...        ...   
28515   0.051860   0.076411   0.105938  -0.112908   0.023364   0.059527   
28516   0.015274   0.025557   0.071350  -0.041278  -0.010443   0.009135   
28517  -0.000337   0.029920   0.065977  -0.035693   0.018643  -0.000174   
28518   0.035765   0.053481   0.131675  -0.043579  -0.011288  -0.002910   
28519   0.004939   0.041245   0.101223  -0.047161   0.007973  -0.006654   

       feature_7  feature_8  feature_9  feature_10  ...  feature_759  \
0      -0.073593  -0.005386  -0.212557    0.099683  ...    -0.085613   
1       0.067114  -0.072412  -0.239192    0.104741  ...    -0.235261   
2      -0.050092  -0.045661  -0.155332    0.117206  ...    -0.146143   
3       0.008974  -0.003277  -0.065046    0.095480  ...    -0.102543   
4      -0.082243  -0.080568  -0.341500    0.142430  ...    -0.253095   
...          ...        ...        ...         ...  ...          ...   
28515  -0.079593  -0.073753  -0.107601    0.104177  ...    -0.134283   
28516  -0.028207  -0.031185  -0.054272    0.029009  ...     0.027247   
28517  -0.051270  -0.022635  -0.041316    0.001559  ...     0.002205   
28518  -0.042682  -0.006227  -0.107985    0.055559  ...     0.004783   
28519  -0.052609  -0.001305  -0.051188    0.023919  ...    -0.016804   

       feature_760  feature_761  feature_762  feature_763  feature_764  \
0         0.116276    -0.009873    -0.168441    -0.085248    -0.096007   
1         0.075487     0.032231    -0.318012    -0.090283    -0.053885   
2         0.113063     0.038823    -0.199746    -0.021524    -0.008411   
3         0.101914     0.043303    -0.202432    -0.071936    -0.023120   
4         0.131308     0.089454    -0.219729    -0.155621    -0.079447   
...            ...          ...          ...          ...          ...   
28515     0.084795     0.041743    -0.128696     0.044550    -0.056263   
28516     0.021062     0.025761    -0.069785     0.025501    -0.018314   
28517     0.030578     0.007359    -0.052741     0.038724    -0.011894   
28518     0.056293     0.019945    -0.112201    -0.011463    -0.035990   
28519     0.037677     0.039298    -0.089907     0.020371    -0.009467   

       feature_765  feature_766  feature_767  feature_768  
0        -0.000766     0.021399    -0.041432     0.094806  
1        -0.010967     0.062209    -0.122958     0.192949  
2        -0.006248     0.031468    -0.056915     0.154731  
3        -0.007812     0.057600    -0.121892     0.072796  
4         0.015316     0.127726    -0.151966     0.169634  
...            ...          ...          ...          ...  
28515     0.004664     0.075623     0.017195     0.240886  
28516    -0.012793     0.000299     0.012029     0.046191  
28517     0.002687     0.008560    -0.004554     0.061891  
28518    -0.018325    -0.010334    -0.020396     0.050323  
28519    -0.001557     0.019498    -0.025880     0.064083  

[28520 rows x 768 columns]

In [7]:
y_train

label_1  label_2  label_3  label_4
0           45      NaN        1        6
1           45      NaN        1        6
2           45      NaN        1        6
3           45      NaN        1        6
4           45      NaN        1        6
...        ...      ...      ...      ...
28515       39     29.0        1        6
28516       39     29.0        1        6
28517       39     29.0        1        6
28518       39     29.0        1        6
28519       39     29.0        1        6

[28520 rows x 4 columns]

In [8]:
valid_df = pd.read_csv('valid.csv')

In [9]:
valid_df.head()

feature_1  feature_2  feature_3  feature_4  feature_5  feature_6  \
0   0.153978   0.503276   0.022196   0.226684   0.466179   0.154995   
1   0.056877   0.261613   0.050610   0.097744   0.322375  -0.011609   
2   0.225439   0.350977  -0.295782   0.280168   0.705114   0.020545   
3   0.288556   0.513905  -0.205246   0.279660   0.469121   0.068339   
4   0.165176   0.312492  -0.217504   0.259960   0.568979  -0.093011   

   feature_7  feature_8  feature_9  feature_10  ...  feature_763  feature_764  \
0   0.195452  -0.288972  -0.297589   -0.008409  ...    -0.503624    -0.056849   
1   0.201331  -0.194763  -0.194228   -0.094267  ...    -0.442110     0.008424   
2   0.390878  -0.322853   0.071575    0.013803  ...    -0.380124    -0.089572   
3   0.131205  -0.338951  -0.270848    0.007799  ...    -0.529678    -0.093194   
4   0.257977  -0.277132  -0.172113   -0.025929  ...    -0.504997    -0.108256   

   feature_765  feature_766  feature_767  feature_768  label_1  label_2  \
0    -0.117077     0.168611     0.373346     0.037188       45      NaN   
1    -0.067920     0.165600     0.333104    -0.270913       45      NaN   
2    -0.023344     0.194312     0.269537    -0.292029       45      NaN   
3    -0.148418     0.405543     0.438906    -0.055119       45      NaN   
4     0.036867     0.267815     0.245041     0.117444       45      NaN   

   label_3  label_4  
0        1        6  
1        1        6  
2        1        6  
3        1        6  
4        1        6  

[5 rows x 772 columns]

In [10]:
valid_df.isnull().sum()

feature_1       0
feature_2       0
feature_3       0
feature_4       0
feature_5       0
               ..
feature_768     0
label_1         0
label_2        14
label_3         0
label_4         0
Length: 772, dtype: int64

In [11]:
y_valid = valid_df[["label_1","label_2","label_3","label_4"]]
X_valid = valid_df.drop(y_valid, axis=1)

In [12]:
X_valid.head()

feature_1  feature_2  feature_3  feature_4  feature_5  feature_6  \
0   0.153978   0.503276   0.022196   0.226684   0.466179   0.154995   
1   0.056877   0.261613   0.050610   0.097744   0.322375  -0.011609   
2   0.225439   0.350977  -0.295782   0.280168   0.705114   0.020545   
3   0.288556   0.513905  -0.205246   0.279660   0.469121   0.068339   
4   0.165176   0.312492  -0.217504   0.259960   0.568979  -0.093011   

   feature_7  feature_8  feature_9  feature_10  ...  feature_759  feature_760  \
0   0.195452  -0.288972  -0.297589   -0.008409  ...     0.058990    -0.112258   
1   0.201331  -0.194763  -0.194228   -0.094267  ...    -0.002137     0.033275   
2   0.390878  -0.322853   0.071575    0.013803  ...    -0.039573    -0.063131   
3   0.131205  -0.338951  -0.270848    0.007799  ...     0.027014    -0.037250   
4   0.257977  -0.277132  -0.172113   -0.025929  ...     0.181833    -0.006872   

   feature_761  feature_762  feature_763  feature_764  feature_765  \
0    -0.128763    -0.243036    -0.503624    -0.056849    -0.117077   
1    -0.042320    -0.012359    -0.442110     0.008424    -0.067920   
2    -0.180942    -0.146527    -0.380124    -0.089572    -0.023344   
3    -0.242656    -0.287592    -0.529678    -0.093194    -0.148418   
4    -0.039907    -0.410866    -0.504997    -0.108256     0.036867   

   feature_766  feature_767  feature_768  
0     0.168611     0.373346     0.037188  
1     0.165600     0.333104    -0.270913  
2     0.194312     0.269537    -0.292029  
3     0.405543     0.438906    -0.055119  
4     0.267815     0.245041     0.117444  

[5 rows x 768 columns]

In [13]:
y_valid.head()

label_1  label_2  label_3  label_4
0       45      NaN        1        6
1       45      NaN        1        6
2       45      NaN        1        6
3       45      NaN        1        6
4       45      NaN        1        6

# KNN Model


In [14]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.model_selection import GridSearchCV
import numpy as np
from sklearn.metrics import accuracy_score

def try_knn(train_x, train_y, valid_x, valid_y):
    grid_params = { 'n_neighbors' : [3,5,7,15],
               'weights' : ['uniform','distance']}
    gs = GridSearchCV(KNeighborsClassifier(), grid_params, verbose = 1, cv=3, n_jobs = -1)
    g_res = gs.fit(train_x, train_y)
    print(f"Best Hyperparameters:",gs.best_params_)
    # Get the best k-NN model with the optimal hyperparameters
    best_knn = gs.best_estimator_
    # Evaluate the best model on the test data
    accuracy = best_knn.score(valid_x, valid_y)
    print(f"Accuracy for KNN {accuracy}")

# SVM Model

In [15]:
from sklearn import svm
from sklearn.metrics import accuracy_score

def try_svm(train_x, train_y, valid_x, valid_y):
    clf = svm.SVC()
    clf.fit(train_x, train_y)
    X_valid_contiguous = np.ascontiguousarray(valid_x)
    y_pred = clf.predict(X_valid_contiguous)
    print(f"Accuracy Score of SVM with default settings (RBF - exponential kernal) ", accuracy_score(valid_y, y_pred))
    
    clf = svm.LinearSVC(dual="auto")
    clf.fit(train_x, train_y)
    X_valid_contiguous = np.ascontiguousarray(valid_x)
    y_pred = clf.predict(X_valid_contiguous)
    print(f"Accuracy Score of Linear SVM (one-vs-the-rest) ", accuracy_score(valid_y, y_pred))
    
    clf = svm.SVC(kernel='linear')
    clf.fit(train_x, train_y)
    X_valid_contiguous = np.ascontiguousarray(valid_x)
    y_pred = clf.predict(X_valid_contiguous)
    print(f"Accuracy Score of SVM with linear kernal function ", accuracy_score(valid_y, y_pred))
    
    clf = svm.SVC(kernel='sigmoid')
    clf.fit(train_x, train_y)
    X_valid_contiguous = np.ascontiguousarray(valid_x)
    y_pred = clf.predict(X_valid_contiguous)
    print(f"Accuracy Score of SVM with sigmoid kernal function ", accuracy_score(valid_y, y_pred))
    
    clf = svm.SVC(kernel='poly',degree =2)
    clf.fit(train_x, train_y)
    X_valid_contiguous = np.ascontiguousarray(valid_x)
    y_pred = clf.predict(X_valid_contiguous)
    print(f"Accuracy Score of SVM with polynomial kernal function with degree 2 ", accuracy_score(valid_y, y_pred))
    
    clf = svm.SVC(kernel='poly',degree =2)
    clf.fit(train_x, train_y)
    X_valid_contiguous = np.ascontiguousarray(valid_x)
    y_pred = clf.predict(X_valid_contiguous)
    print(f"Accuracy Score of SVM with polynomial kernal function with degree 3 ", accuracy_score(valid_y, y_pred))

# xgBoost Model

In [16]:
import xgboost as xgb
from sklearn.preprocessing import LabelEncoder

def try_xgBoost(train_x, train_y, vaid_x, valid_y):
    # Create an XGBoost classifier for multi-class classification
    clf = xgb.XGBClassifier(
    objective='multi:softmax',  # Set the objective for multi-class classification
    num_class=len(np.unique(train_y)),  # Number of classes
    random_state=42
    )
    le = LabelEncoder()
    train_y = le.fit_transform(train_y)
    clf.fit(train_x, train_y)
    y_pred = clf.predict(vaid_x)
    y_pred = le.inverse_transform(y_pred)
    print(f"Accuracy Score of XG Boost", accuracy_score(valid_y, y_pred))

# Rondom Forest Model

In [17]:
from sklearn.ensemble import RandomForestClassifier

def try_random_forest(train_x, train_y, vaid_x, valid_y):
    clf = RandomForestClassifier(n_estimators=100, random_state=42)
    clf.fit(train_x, train_y)
    y_pred = clf.predict(vaid_x)
    print(f"Accuracy Score of Random Forest", accuracy_score(valid_y, y_pred))

# Logistic Regression Model

In [18]:
from sklearn.linear_model import LogisticRegression

def try_logistic(train_x, train_y, vaid_x, valid_y):
    for i in ["lbfgs","newton-cg","sag","saga"]:
        logistic_regression = LogisticRegression(multi_class='multinomial', solver='lbfgs')
        logistic_regression.fit(train_x,train_y)
        y_pred = logistic_regression.predict(vaid_x)
        print(f"Accuracy Score of Logistic Regression with solver {i}", accuracy_score(valid_y, y_pred))

# Predict a probability for the TARGET variable

In [19]:
test_df=pd.read_csv('test.csv')
test_df.head()

ID  feature_1  feature_2  feature_3  feature_4  feature_5  feature_6  \
0   1   0.124623   0.196628   0.257004  -0.156045  -0.054916   0.006071   
1   2   0.109655   0.170158   0.227644  -0.127088  -0.044476  -0.046852   
2   3   0.014854   0.030051   0.115092  -0.017179   0.002720  -0.011692   
3   4   0.196893   0.113314   0.352175  -0.108499  -0.064472  -0.073239   
4   5   0.033004   0.013373   0.124001  -0.016143   0.010120   0.010635   

   feature_7  feature_8  feature_9  ...  feature_759  feature_760  \
0  -0.035149  -0.092019  -0.196302  ...    -0.221466     0.140292   
1  -0.090026  -0.061321  -0.227288  ...    -0.204930     0.110203   
2  -0.078855  -0.042991  -0.096283  ...    -0.032937     0.075821   
3  -0.086402   0.008671  -0.342217  ...    -0.255167     0.096579   
4  -0.055789  -0.036282  -0.059422  ...    -0.035814     0.093764   

   feature_761  feature_762  feature_763  feature_764  feature_765  \
0     0.123622    -0.175572    -0.107030    -0.087621    -0.026501   
1     0.085665    -0.286787    -0.113195    -0.057312    -0.055680   
2     0.030987    -0.149850    -0.003155    -0.010207    -0.001427   
3     0.069413    -0.215386    -0.075168    -0.035071    -0.023375   
4     0.027321    -0.116009     0.010096    -0.042293     0.005347   

   feature_766  feature_767  feature_768  
0     0.139337    -0.083030     0.059507  
1     0.143939    -0.045760     0.106113  
2     0.000934    -0.017069     0.048123  
3     0.067768    -0.181530     0.174444  
4     0.007722    -0.007731     0.058799  

[5 rows x 769 columns]

In [20]:
X_test = test_df.drop(["ID"], axis=1)

In [21]:
X_test.head()

feature_1  feature_2  feature_3  feature_4  feature_5  feature_6  \
0   0.124623   0.196628   0.257004  -0.156045  -0.054916   0.006071   
1   0.109655   0.170158   0.227644  -0.127088  -0.044476  -0.046852   
2   0.014854   0.030051   0.115092  -0.017179   0.002720  -0.011692   
3   0.196893   0.113314   0.352175  -0.108499  -0.064472  -0.073239   
4   0.033004   0.013373   0.124001  -0.016143   0.010120   0.010635   

   feature_7  feature_8  feature_9  feature_10  ...  feature_759  feature_760  \
0  -0.035149  -0.092019  -0.196302    0.077971  ...    -0.221466     0.140292   
1  -0.090026  -0.061321  -0.227288    0.066863  ...    -0.204930     0.110203   
2  -0.078855  -0.042991  -0.096283    0.042701  ...    -0.032937     0.075821   
3  -0.086402   0.008671  -0.342217    0.104941  ...    -0.255167     0.096579   
4  -0.055789  -0.036282  -0.059422    0.060278  ...    -0.035814     0.093764   

   feature_761  feature_762  feature_763  feature_764  feature_765  \
0     0.123622    -0.175572    -0.107030    -0.087621    -0.026501   
1     0.085665    -0.286787    -0.113195    -0.057312    -0.055680   
2     0.030987    -0.149850    -0.003155    -0.010207    -0.001427   
3     0.069413    -0.215386    -0.075168    -0.035071    -0.023375   
4     0.027321    -0.116009     0.010096    -0.042293     0.005347   

   feature_766  feature_767  feature_768  
0     0.139337    -0.083030     0.059507  
1     0.143939    -0.045760     0.106113  
2     0.000934    -0.017069     0.048123  
3     0.067768    -0.181530     0.174444  
4     0.007722    -0.007731     0.058799  

[5 rows x 768 columns]

In [22]:
X_test.isnull().sum()

feature_1      0
feature_2      0
feature_3      0
feature_4      0
feature_5      0
              ..
feature_764    0
feature_765    0
feature_766    0
feature_767    0
feature_768    0
Length: 768, dtype: int64

# Label_1


# Use the SVM Model for label_1

In [24]:
try_svm(X_train,y_train["label_1"],X_valid,y_valid["label_1"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with default settings (RBF - exponential kernal)  0.016


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but LinearSVC was fitted with feature names
  warnings.warn(


Accuracy Score of Linear SVM (one-vs-the-rest)  0.06266666666666666


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with linear kernal function  0.176


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with sigmoid kernal function  0.07733333333333334


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with polynomial kernal function with degree 2  0.212


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with polynomial kernal function with degree 3  0.212


# Use the KNN Model for label_1

In [27]:
try_knn(X_train,y_train["label_1"],X_valid,y_valid["label_1"])

Fitting 3 folds for each of 8 candidates, totalling 24 fits
Best Hyperparameters: {'n_neighbors': 5, 'weights': 'distance'}
Accuracy for KNN 0.24266666666666667


In [28]:
try_xgBoost(X_train,y_train["label_1"],X_valid,y_valid["label_1"])

Accuracy Score of XG Boost 0.06933333333333333


In [29]:
try_random_forest(X_train,y_train["label_1"],X_valid,y_valid["label_1"])

Accuracy Score of Random Forest 0.06133333333333333


In [30]:
try_logistic(X_train,y_train["label_1"],X_valid,y_valid["label_1"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver lbfgs 0.15066666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver newton-cg 0.15066666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver sag 0.15066666666666667
Accuracy Score of Logistic Regression with solver saga 0.15066666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


# For label_1 we get best accuracy(0.24266) when using the KNN Model
So we can use this method to predict the values for test data set.

In [37]:
k = 5
knn_model = KNeighborsClassifier(n_neighbors=k)
knn_model.fit(X_train, y_train["label_1"])
X_test_contiguous = np.ascontiguousarray(X_test)
y_pred_label1 = knn_model.predict(X_test_contiguous)
y_pred_label1

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but KNeighborsClassifier was fitted with feature names
  warnings.warn(


array([26, 18, 16,  7, 58, 46,  7, 22, 29, 26, 33, 32, 54, 51, 17, 48, 23,
        2, 43, 42, 11, 35, 39,  2, 21, 13, 28, 55,  2, 51, 55, 16,  1, 56,
       30, 22, 31,  6, 24, 18,  5, 47,  6, 46, 59, 25, 31, 18,  5, 32, 25,
       56, 39, 37, 32, 40,  5, 52, 28, 34,  2, 47, 55, 24, 23, 18, 20,  6,
       36, 13, 22, 26, 44, 46, 55, 36, 49, 25, 57, 12, 52, 38, 23,  4, 33,
        1, 14, 19, 12, 39, 36, 43, 34, 32, 54, 46, 27,  5,  7, 42, 19, 26,
       31, 36, 53, 10, 12, 38, 38,  2, 57,  4,  7, 50, 46, 46, 17, 34, 30,
       40, 16, 40,  8, 22, 38, 15, 13,  9, 52,  5, 40, 13, 10, 55, 28, 25,
       42, 57, 16, 27, 51,  5, 51, 27, 32, 33, 16,  3, 11, 23, 42, 37, 47,
       50,  3, 48, 19, 20, 15,  8,  5, 18,  5, 28, 46, 23, 50, 40,  1, 32,
       24, 57, 42, 16, 11, 35, 34, 52, 41, 20,  7,  8, 21,  4, 37, 55, 20,
       28, 23, 35, 46, 52, 30, 27,  1,  7, 28,  5,  3,  1, 17, 60,  3,  3,
       22,  9, 24, 25, 17, 26, 19, 36, 36, 21,  1, 58, 31, 47, 38,  6, 50,
       39, 23,  5, 48, 22

# Label_2

In [23]:
df_label_2 = df.dropna(subset=['label_2'])
df_label_2.isnull().sum()

feature_1      0
feature_2      0
feature_3      0
feature_4      0
feature_5      0
              ..
feature_768    0
label_1        0
label_2        0
label_3        0
label_4        0
Length: 772, dtype: int64

In [24]:
y_train_l2 = df_label_2[["label_1","label_2","label_3","label_4"]]
X_train_l2 = df_label_2.drop(y_train_l2, axis=1)

In [25]:
y_train_l2.head()

label_1  label_2  label_3  label_4
480        5     25.0        1        6
481        5     25.0        1        6
482        5     25.0        1        6
483        5     25.0        1        6
484        5     25.0        1        6

In [26]:
valid_df_label_2 = valid_df.dropna(subset=['label_2'])
valid_df_label_2.isnull().sum()

feature_1      0
feature_2      0
feature_3      0
feature_4      0
feature_5      0
              ..
feature_768    0
label_1        0
label_2        0
label_3        0
label_4        0
Length: 772, dtype: int64

In [27]:
y_valid_l2 = valid_df_label_2[["label_1","label_2","label_3","label_4"]]
X_valid_l2 = valid_df_label_2.drop(y_train_l2, axis=1)

In [28]:
y_valid_l2.head()

label_1  label_2  label_3  label_4
14        5     25.0        1        6
15        5     25.0        1        6
16        5     25.0        1        6
17        5     25.0        1        6
18        5     25.0        1        6

# Use the SVM Model for label_2

In [ ]:
try_svm(X_train_l2,y_train_l2["label_2"],X_valid_l2,y_valid_l2["label_2"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with default settings (RBF - exponential kernal)  0.12092391304347826


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but LinearSVC was fitted with feature names
  warnings.warn(


Accuracy Score of Linear SVM (one-vs-the-rest)  0.13994565217391305


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with linear kernal function  0.13179347826086957


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with sigmoid kernal function  0.06521739130434782


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with polynomial kernal function with degree 2  0.1480978260869565


# Use the KNN Model for label_2

In [31]:
try_knn(X_train_l2,y_train_l2["label_2"],X_valid_l2,y_valid_l2["label_2"])

Fitting 3 folds for each of 8 candidates, totalling 24 fits
Best Hyperparameters: {'n_neighbors': 3, 'weights': 'distance'}
Accuracy for KNN 0.27445652173913043


In [29]:
try_xgBoost(X_train_l2,y_train_l2["label_2"],X_valid_l2,y_valid_l2["label_2"])

Accuracy Score of XG Boost 0.11413043478260869


In [30]:
try_random_forest(X_train_l2,y_train_l2["label_2"],X_valid_l2,y_valid_l2["label_2"])

Accuracy Score of Random Forest 0.12092391304347826


In [31]:
try_logistic(X_train_l2,y_train_l2["label_2"],X_valid_l2,y_valid_l2["label_2"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver lbfgs 0.17119565217391305


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver newton-cg 0.17119565217391305


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver sag 0.17119565217391305
Accuracy Score of Logistic Regression with solver saga 0.17119565217391305


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


For label_4 we get best accuracy(0.27445) when using the KNN model. 
So we can use this methods to predict the values for test data set. 

In [33]:
k = 3
knn_model = KNeighborsClassifier(n_neighbors=k)
knn_model.fit(X_train_l2, y_train_l2["label_2"])
X_test_contiguous = np.ascontiguousarray(X_test)
y_pred_label2 = knn_model.predict(X_test_contiguous)
y_pred_label2

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but KNeighborsClassifier was fitted with feature names
  warnings.warn(


array([22., 25., 30., 27., 29., 23., 27., 33., 23., 22., 26., 23., 27.,
       26., 23., 26., 28., 25., 31., 29., 33., 24., 29., 25., 26., 22.,
       28., 23., 25., 26., 23., 30., 30., 24., 28., 23., 26., 25., 26.,
       25., 23., 23., 25., 30., 31., 22., 26., 25., 25., 23., 22., 24.,
       29., 27., 23., 26., 24., 34., 28., 25., 23., 23., 23., 26., 28.,
       25., 25., 25., 22., 23., 33., 22., 61., 30., 26., 22., 26., 22.,
       27., 26., 34., 32., 28., 31., 26., 26., 31., 23., 26., 29., 22.,
       31., 25., 23., 27., 30., 31., 25., 27., 29., 23., 22., 26., 22.,
       24., 36., 26., 32., 32., 25., 27., 26., 26., 24., 30., 30., 26.,
       25., 28., 26., 26., 26., 41., 33., 32., 28., 24., 35., 34., 25.,
       26., 27., 36., 23., 28., 22., 29., 27., 30., 31., 26., 23., 22.,
       31., 23., 26., 30., 23., 33., 24., 29., 27., 23., 24., 31., 26.,
       23., 25., 28., 41., 25., 25., 25., 28., 30., 26., 24., 26., 26.,
       23., 26., 27., 29., 30., 33., 24., 25., 34., 28., 23., 27

# Label_3


# Use the SVM Model for label_3

In [32]:
try_svm(X_train,y_train["label_3"],X_valid,y_valid["label_3"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with default settings (RBF - exponential kernal)  0.8106666666666666


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but LinearSVC was fitted with feature names
  warnings.warn(


Accuracy Score of Linear SVM (one-vs-the-rest)  0.8533333333333334


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with linear kernal function  0.8653333333333333


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with sigmoid kernal function  0.7306666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with polynomial kernal function with degree 2  0.8706666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with polynomial kernal function with degree 3  0.8706666666666667


# Use the KNN Model for label_3

In [33]:
try_knn(X_train,y_train["label_3"],X_valid,y_valid["label_3"])

Fitting 3 folds for each of 8 candidates, totalling 24 fits
Best Hyperparameters: {'n_neighbors': 7, 'weights': 'uniform'}
Accuracy for KNN 0.8266666666666667


In [42]:
try_xgBoost(X_train,y_train["label_3"],X_valid,y_valid["label_3"])

Accuracy Score of XG Boost 0.8133333333333334


In [43]:
try_random_forest(X_train,y_train["label_3"],X_valid,y_valid["label_3"])

Accuracy Score of Random Forest 0.47333333333333333


In [32]:
try_logistic(X_train,y_train["label_3"],X_valid,y_valid["label_3"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver lbfgs 0.8773333333333333


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver newton-cg 0.8773333333333333


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver sag 0.8773333333333333
Accuracy Score of Logistic Regression with solver saga 0.8773333333333333


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


For label_3 we get best accuracy(0.87) when using the SVM with polynomial kernal function and Logistic Regression models.
So we can use any of these methods to predict the values for test data set. I am going to use SVM with polynomial kernal.

In [35]:
clf = svm.SVC(kernel='poly',degree =2)
clf.fit(X_train, y_train["label_3"])
X_test_contiguous = np.ascontiguousarray(X_test)
y_pred_label3 = clf.predict(X_test_contiguous)
y_pred_label3

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


array([0, 1, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1,
       1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1,
       0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1,
       1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1,
       0, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 1, 0, 1, 1, 1,
       0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1,
       1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1,
       1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 1, 1, 1, 1, 0, 1,
       1, 0, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1,
       1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0, 1, 1,
       1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0,
       1, 1, 0, 1, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1,
       1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,

# Label_4

# Use the SVM Model for label_4

In [34]:
try_svm(X_train,y_train["label_4"],X_valid,y_valid["label_4"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with default settings (RBF - exponential kernal)  0.7093333333333334


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but LinearSVC was fitted with feature names
  warnings.warn(


Accuracy Score of Linear SVM (one-vs-the-rest)  0.052


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with linear kernal function  0.06266666666666666


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with sigmoid kernal function  0.396


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with polynomial kernal function with degree 2  0.10666666666666667


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


Accuracy Score of SVM with polynomial kernal function with degree 3  0.10666666666666667


# Use the KNN Model for label_4

In [35]:
try_knn(X_train,y_train["label_4"],X_valid,y_valid["label_4"])

Fitting 3 folds for each of 8 candidates, totalling 24 fits
Best Hyperparameters: {'n_neighbors': 7, 'weights': 'distance'}
Accuracy for KNN 0.676


In [37]:
try_xgBoost(X_train,y_train["label_4"],X_valid,y_valid["label_4"])

Accuracy Score of XG Boost 0.33066666666666666


In [38]:
try_random_forest(X_train,y_train["label_4"],X_valid,y_valid["label_4"])

Accuracy Score of Random Forest 0.6866666666666666


In [41]:
try_logistic(X_train,y_train["label_4"],X_valid,y_valid["label_4"])

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver lbfgs 0.10933333333333334


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver newton-cg 0.10933333333333334


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Accuracy Score of Logistic Regression with solver sag 0.10933333333333334
Accuracy Score of Logistic Regression with solver saga 0.10933333333333334


/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


For label_4 we get best accuracy(0.70933) when using the SVM with default settings.
So we can use this methods to predict the values for test data set. 

In [40]:
clf = svm.SVC()
clf.fit(X_train, y_train["label_4"])
X_test_contiguous = np.ascontiguousarray(X_test)
y_pred_label4 = clf.predict(X_test_contiguous)
y_pred_label4    

/home/sakeerthan/.local/lib/python3.8/site-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but SVC was fitted with feature names
  warnings.warn(


array([ 2,  8,  6,  6,  6,  6,  6,  6,  6,  2,  6,  6,  6,  6,  6,  6,  6,
        6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
        6,  6,  6,  6,  6,  6,  4,  6,  6,  6,  6,  6,  6,  6,  6,  0,  6,
        6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
        6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
        6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  0,  6,  2,
        6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
        6,  6,  6,  6,  6,  6,  6,  6, 11,  5,  6,  6,  6,  6,  6,  6,  1,
        6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
        6,  6,  6,  4,  6,  9,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
        6,  6,  6,  6,  6,  6,  6,  5, 10,  6,  6,  6,  6,  6,  6,  6,  6,
        6,  6,  2,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,  6,
        6, 11,  6,  6,  6,  2,  4,  6,  6,  6,  6,  6,  6,  3,  6,  6,  6,
        6,  6,  6,  6,  6

# Final output

In [41]:
final_df = pd.concat([pd.Series(test_df['ID']), pd.Series(y_pred_label1, name='label_1'), pd.Series(y_pred_label2, name='label_2'), pd.Series(y_pred_label3, name='label_3'), pd.Series(y_pred_label4, name='label_4')], axis=1)
output_file_name = "190541R_lab_2_II.csv"
final_df.to_csv(output_file_name, index=False)
